

<!DOCTYPE html>


<html lang="en" >

  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

    <title>How does MICOM model communities &#8212; version 0.32.4</title>
  
  
  
  <script data-cfasync="false">
    document.documentElement.dataset.mode = localStorage.getItem("mode") || "";
    document.documentElement.dataset.theme = localStorage.getItem("theme") || "light";
  </script>
  
  <!-- Loaded before other Sphinx assets -->
  <link href="_static/styles/theme.css?digest=12da95d707ffb74b382d" rel="stylesheet" />
<link href="_static/styles/bootstrap.css?digest=12da95d707ffb74b382d" rel="stylesheet" />
<link href="_static/styles/pydata-sphinx-theme.css?digest=12da95d707ffb74b382d" rel="stylesheet" />

  
  <link href="_static/vendor/fontawesome/6.1.2/css/all.min.css?digest=12da95d707ffb74b382d" rel="stylesheet" />
  <link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-solid-900.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-brands-400.woff2" />
<link rel="preload" as="font" type="font/woff2" crossorigin href="_static/vendor/fontawesome/6.1.2/webfonts/fa-regular-400.woff2" />

    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" href="_static/styles/sphinx-book-theme.css?digest=14f4ca6b54d191a8c7657f6c759bf11a5fb86285" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/graphviz.css" />
    <link rel="stylesheet" type="text/css" href="_static/nbsphinx-code-cells.css" />
  
  <!-- Pre-loaded scripts that we'll load fully later -->
  <link rel="preload" as="script" href="_static/scripts/bootstrap.js?digest=12da95d707ffb74b382d" />
<link rel="preload" as="script" href="_static/scripts/pydata-sphinx-theme.js?digest=12da95d707ffb74b382d" />

    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/scripts/sphinx-book-theme.js?digest=5a5c038af52cf7bc1a1ec88eea08e6366ee68824"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    <script>DOCUMENTATION_OPTIONS.pagename = 'logic';</script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Installing MICOM" href="installing.html" />
    <link rel="prev" title="Welcome" href="index.html" />
  <meta name="viewport" content="width=device-width, initial-scale=1"/>
  <meta name="docsearch:language" content="en"/>
  </head>
  
  
  <body data-bs-spy="scroll" data-bs-target=".bd-toc-nav" data-offset="180" data-bs-root-margin="0px 0px -60%" data-default-mode="">

  
  
  <a class="skip-link" href="#main-content">Skip to main content</a>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__primary"
          id="__primary"/>
  <label class="overlay overlay-primary" for="__primary"></label>
  
  <input type="checkbox"
          class="sidebar-toggle"
          name="__secondary"
          id="__secondary"/>
  <label class="overlay overlay-secondary" for="__secondary"></label>
  
  <div class="search-button__wrapper">
    <div class="search-button__overlay"></div>
    <div class="search-button__search-container">
<form class="bd-search d-flex align-items-center"
      action="search.html"
      method="get">
  <i class="fa-solid fa-magnifying-glass"></i>
  <input type="search"
         class="form-control"
         name="q"
         id="search-input"
         placeholder="Search..."
         aria-label="Search..."
         autocomplete="off"
         autocorrect="off"
         autocapitalize="off"
         spellcheck="false"/>
  <span class="search-button__kbd-shortcut"><kbd class="kbd-shortcut__modifier">Ctrl</kbd>+<kbd>K</kbd></span>
</form></div>
  </div>
  
    <nav class="bd-header navbar navbar-expand-lg bd-navbar">
    </nav>
  
  <div class="bd-container">
    <div class="bd-container__inner bd-page-width">
      
      <div class="bd-sidebar-primary bd-sidebar">
        

  
  <div class="sidebar-header-items sidebar-primary__section">
    
    
    
    
  </div>
  
    <div class="sidebar-primary-items__start sidebar-primary__section">
        <div class="sidebar-primary-item">
  

<a class="navbar-brand logo" href="index.html">
  
  
  
  
    
    
      
    
    
    <img src="_static/micom.png" class="logo__image only-light" alt="Logo image"/>
    <script>document.write(`<img src="_static/micom.png" class="logo__image only-dark" alt="Logo image"/>`);</script>
  
  
</a></div>
        <div class="sidebar-primary-item"><nav class="bd-links" id="bd-docs-nav" aria-label="Main">
    <div class="bd-toc-item navbar-nav active">
        <ul class="current nav bd-sidenav">
<li class="toctree-l1 current active"><a class="current reference internal" href="#">Methods used by MICOM</a></li>
<li class="toctree-l1"><a class="reference internal" href="installing.html">Installing MICOM</a></li>
<li class="toctree-l1"><a class="reference internal" href="high_level.html">MICOM workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="viz.html">Visualizations</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflows.html">Writing your own workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="community.html">Single Model API</a></li>
<li class="toctree-l1"><a class="reference internal" href="growth_fluxes.html">Growth rates and fluxes</a></li>
<li class="toctree-l1"><a class="reference internal" href="media.html">Growth media</a></li>
<li class="toctree-l1"><a class="reference internal" href="taxa_knockouts.html">Knockouts</a></li>
<li class="toctree-l1"><a class="reference internal" href="elasticities.html">Intervention studies</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorials.html">Tutorials and Courses</a></li>
<li class="toctree-l1"><a class="reference internal" href="NEWS.html">Release Notes</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="autoapi/index.html">API Reference</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-1"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l2 has-children"><a class="reference internal" href="autoapi/micom/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom</span></code></a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-2"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/data/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.data</span></code></a></li>
<li class="toctree-l3 has-children"><a class="reference internal" href="autoapi/micom/viz/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz</span></code></a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-3"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/core/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz.core</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/exchanges/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz.exchanges</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/growth/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz.growth</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/prediction/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz.prediction</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/tradeoff/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz.tradeoff</span></code></a></li>
</ul>
</li>
<li class="toctree-l3 has-children"><a class="reference internal" href="autoapi/micom/workflows/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows</span></code></a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" type="checkbox"/><label class="toctree-toggle" for="toctree-checkbox-4"><i class="fa-solid fa-chevron-down"></i></label><ul>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/build/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.build</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/core/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.core</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/db_media/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.db_media</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/grow/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.grow</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/media/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.media</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/tradeoff/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.tradeoff</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/algorithms/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.algorithms</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/annotation/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.annotation</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/community/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.community</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/db/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.db</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/deps/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.deps</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/duality/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.duality</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/elasticity/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.elasticity</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/logger/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.logger</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/measures/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.measures</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/media/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.media</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/optcom/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.optcom</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/problems/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.problems</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/qiime_formats/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.qiime_formats</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/solution/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.solution</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/stats/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.stats</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/taxonomy/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.taxonomy</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/util/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.util</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>

    </div>
</nav></div>
    </div>
  
  
  <div class="sidebar-primary-items__end sidebar-primary__section">
  </div>
  
  <div id="rtd-footer-container"></div>


      </div>
      
      <main id="main-content" class="bd-main">
        
        

<div class="sbt-scroll-pixel-helper"></div>

          <div class="bd-content">
            <div class="bd-article-container">
              
              <div class="bd-header-article">
<div class="header-article-items header-article__inner">
  
    <div class="header-article-items__start">
      
        <div class="header-article-item"><label class="sidebar-toggle primary-toggle btn btn-sm" for="__primary" title="Toggle primary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
  <span class="fa-solid fa-bars"></span>
</label></div>
      
    </div>
  
  
    <div class="header-article-items__end">
      
        <div class="header-article-item">

<div class="article-header-buttons">





<div class="dropdown dropdown-download-buttons">
  <button class="btn dropdown-toggle" type="button" data-bs-toggle="dropdown" aria-expanded="false" aria-label="Download this page">
    <i class="fas fa-download"></i>
  </button>
  <ul class="dropdown-menu">
      
      
      
      <li><a href="_sources/logic.ipynb" target="_blank"
   class="btn btn-sm btn-download-source-button dropdown-item"
   title="Download source file"
   data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file"></i>
  </span>
<span class="btn__text-container">.ipynb</span>
</a>
</li>
      
      
      
      
      <li>
<button onclick="window.print()"
  class="btn btn-sm btn-download-pdf-button dropdown-item"
  title="Print to PDF"
  data-bs-placement="left" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-file-pdf"></i>
  </span>
<span class="btn__text-container">.pdf</span>
</button>
</li>
      
  </ul>
</div>




<button onclick="toggleFullScreen()"
  class="btn btn-sm btn-fullscreen-button"
  title="Fullscreen mode"
  data-bs-placement="bottom" data-bs-toggle="tooltip"
>
  

<span class="btn__icon-container">
  <i class="fas fa-expand"></i>
  </span>

</button>


<script>
document.write(`
  <button class="theme-switch-button btn btn-sm btn-outline-primary navbar-btn rounded-circle" title="light/dark" aria-label="light/dark" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="theme-switch" data-mode="light"><i class="fa-solid fa-sun"></i></span>
    <span class="theme-switch" data-mode="dark"><i class="fa-solid fa-moon"></i></span>
    <span class="theme-switch" data-mode="auto"><i class="fa-solid fa-circle-half-stroke"></i></span>
  </button>
`);
</script>

<script>
document.write(`
  <button class="btn btn-sm navbar-btn search-button search-button__button" title="Search" aria-label="Search" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <i class="fa-solid fa-magnifying-glass"></i>
  </button>
`);
</script>
<label class="sidebar-toggle secondary-toggle btn btn-sm" for="__secondary"title="Toggle secondary sidebar" data-bs-placement="bottom" data-bs-toggle="tooltip">
    <span class="fa-solid fa-list"></span>
</label>
</div></div>
      
    </div>
  
</div>
</div>
              
              

<div id="jb-print-docs-body" class="onlyprint">
    <h1>How does MICOM model communities</h1>
    <!-- Table of contents -->
    <div id="print-main-content">
        <div id="jb-print-toc">
            
            <div>
                <h2> Contents </h2>
            </div>
            <nav aria-label="Page">
                <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#Exchanges-and-community-growth-rate">Exchanges and community growth rate</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#Steady-states-in-the-community">Steady states in the community</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#Cooperative-tradeoff">Cooperative tradeoff</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#Numerical-stability-and-solvers">Numerical stability and solvers</a></li>
</ul>
            </nav>
        </div>
    </div>
</div>

              
                
<div id="searchbox"></div>
                <article class="bd-article" role="main">
                  
  <section id="How-does-MICOM-model-communities">
<h1>How does MICOM model communities<a class="headerlink" href="#How-does-MICOM-model-communities" title="Permalink to this heading">#</a></h1>
<p><img alt="image0" src="_images/summary.png" /></p>
<p>There are many frameworks for microbial modeling and all make their own specific assumptions in order to fit microbial communities into a mathematical representation. This is an overview about the assumptions <code class="docutils literal notranslate"><span class="pre">MICOM</span></code> makes and how it translates the community into mathematical terms. As it happens all formulations here are similar to formulation used in the <a class="reference external" href="https://doi.org/10.1371/journal.pcbi.1002363">OptCom</a> and <a class="reference external" href="https://doi.org/10.1371/journal.pcbi.1005539">SteadyCom</a> papers.</p>
<p>In general <code class="docutils literal notranslate"><span class="pre">MICOM</span></code> makes several assumptions to infer growth rates and fluxes in a microbial community. The most important ones being:</p>
<ol class="arabic simple">
<li><p>Growth rates and relative abundances in the community are in steady state.</p></li>
<li><p>Most taxa that are present in a biological sample should be able to grow (have a growth rate &gt;0).</p></li>
<li><p>Relative DNA abundances can be used as a proxy for dry weight taxa abundances.</p></li>
</ol>
<p>Different to SteadyCom, MICOM does not assume that all taxa have the same growth rate and allows for taxa-specific dilution rates.</p>
<section id="Exchanges-and-community-growth-rate">
<h2>Exchanges and community growth rate<a class="headerlink" href="#Exchanges-and-community-growth-rate" title="Permalink to this heading">#</a></h2>
<p>One of the things that can easily overlooked when using FBA for communities is that growth rates and fluxes are usually given in unit mass/(abundance * time), for instance mmmol/(gDW * h). Thus, all fluxes are relative to the abundance of the single bacteria they describe. However, in a community all bacteria might have different abundances so we have to take care to balance fluxes. For instance let us take a system containing only one bacteria <span class="math notranslate nohighlight">\(i\)</span> which imports metabolite X from the
external medium. In the medium metabolite X enters the system with an unscaled flux <span class="math notranslate nohighlight">\(v^m_x\)</span> which has units mmmol/h. Within the bacteria the metabolite is consumed with the sclaed flux <span class="math notranslate nohighlight">\(v^i_x\)</span> which has units mmol/(gDW * h). <span class="math notranslate nohighlight">\(v^i_x\)</span> described the flux that can be realized by 1 gDW of the respective bacteria. However, the abundance for bacteria <span class="math notranslate nohighlight">\(i\)</span>, <span class="math notranslate nohighlight">\(b_i\)</span> (in gDW), might be different than that. In order to have balanced fluxes we have to enforce that
<span class="math notranslate nohighlight">\(v^m_x = b_i \cdot v^i_x\)</span> (overall influx equals overall consumption). If many bacteria import the respective metabolite we thus need to enforce</p>
<div class="math notranslate nohighlight">
\[v^m_x = \sum_i b_i \cdot v^i_x\]</div>
<p>The actual abundances <span class="math notranslate nohighlight">\(b_i\)</span> are usually not known, but we can divide the equation by the total bacterial abundance <span class="math notranslate nohighlight">\(B = \sum_i b_i\)</span> and obtain</p>
<div class="math notranslate nohighlight">
\[\tilde{v}^m_x = v^m_x/B = \sum_i b_i/B \cdot v^i_x\]</div>
<p><span class="math notranslate nohighlight">\(\tilde{v}^m_x\)</span> is now a scaled flux in the medium relative to the overall bacterial biomass in the community. The relative abundances <span class="math notranslate nohighlight">\(\tilde{b}_i = b_i/B\)</span> can be taken from metagenomic studies such as 16S rRNA quantities. This is how <code class="docutils literal notranslate"><span class="pre">MICOM</span></code> uses abundance data.</p>
<p>In a similar manner the unscaled community growth rate (total biomass production) is given by</p>
<div class="math notranslate nohighlight">
\[v_{biomass} = \sum_i b_i\cdot v^i_{biomass}\]</div>
<p>and dividing by the total biomass <span class="math notranslate nohighlight">\(B\)</span> yields</p>
<div class="math notranslate nohighlight">
\[\tilde{v}_{biomass} = \sum_i b_i/B v^i_{biomass}\]</div>
<p>where <span class="math notranslate nohighlight">\(\tilde{v}_{biomass}\)</span> now again is a scaled biomass flux relative to a total community biomass of 1 gDW. In order to adjust growth rate predictions from <code class="docutils literal notranslate"><span class="pre">MICOM</span></code> to your sample you would have to divide those growth rates by the overall weight of microbiota in your sample. For instance <a class="reference external" href="https://dx.doi.org/10.1371%2Fjournal.pbio.1002533">a recent publication</a> estimates that an average human has 200g of bacteria in the gut.</p>
</section>
<section id="Steady-states-in-the-community">
<h2>Steady states in the community<a class="headerlink" href="#Steady-states-in-the-community" title="Permalink to this heading">#</a></h2>
<p><code class="docutils literal notranslate"><span class="pre">MICOM</span></code> has be designed with the gut microbiota in mind. One of the major problems when trying to apply flux balance analysis (FBA) to microbial community data especially metagenomic data are paradoxical assumptions about community growth. Flux balance analysis usually assumes a maximization of the growth rate, or at least the realization of one particular growth rate <span class="math notranslate nohighlight">\(\mu\)</span>, however metagenomic experiments usually only quantify the microbial compositions at one particular time point,
assuming that the microbial abundance does not change. Those two assumptions are not compatible per sé. If members of the community grow with a constant rate they will accumulate exponentially over time, however we know that this is not the case in systems as the intestine (otherwise we would probably explode due to an overpopulation of bacteria and fungi). This disagreement can be aleviated by accounting for dilution of the microbiota. For instance in the gut bacteria are constantly removed in
small amounts by death (when arriving at their specific life span) and in larger amounts by defecation. The sum of all processes removing bacteria from the system is what we call dilution here. In <code class="docutils literal notranslate"><span class="pre">MICOM</span></code> we assume the follwing about the dilution process:</p>
<ol class="arabic simple">
<li><p>It is relative to the bacterial abundance (the more you have in your system the more is removed by dilution)</p></li>
<li><p>It may be specific to the bacterial strain</p></li>
<li><p>It may specific to the sample</p></li>
</ol>
<ol class="arabic simple">
<li><p>is known to be true for the gut microbiome since we know that a higher concentration of a bacteria in the gut is usually associated with a higher concentration in stool samples (which is the major dilution contributor). This is similar to <a class="reference external" href="https://doi.org/10.1371/journal.pcbi.1005539.s016">SteadyCom</a> but does not assume that the dilution rate is constant across all taxa. (2) is based on the observation that bacteria may have distinct spatial arrangements which make it easier or hader to
be diluted. (3) is based on the assumption that the respective systems may be different (no gut is the same :D), however it is one of the assumptions we are currently trying to validate. As a consequence we assume that the abundance for bacteria <span class="math notranslate nohighlight">\(i\)</span>, <span class="math notranslate nohighlight">\(b_i\)</span> occurs with a growth rate <span class="math notranslate nohighlight">\(\mu_i\)</span> and balances with an arbitrary dilution process <span class="math notranslate nohighlight">\(d_i(t) = d_i(b_i)\)</span> as</p></li>
</ol>
<div class="math notranslate nohighlight">
\[\frac{d b_i(t)}{dt} = \mu_i b_i(t) - d_i(t)\]</div>
<p>As we can see absolute bacterial abundance can only be in steady state if growth and dilution are balanced. Additionally one could also formulate the problem in terms of relative concentrations to the total community abundance <span class="math notranslate nohighlight">\(B = \sum_i b_i\)</span> using the quotient rule, which yields:</p>
<div class="math notranslate nohighlight">
\[\frac{d \tilde{b}_i}{dt} = \frac{d b_i/B}{dt} = \mu_i\tilde{b}_i - \frac{d_i(t)}{B} - \cdot \tilde{b}_i\sum_k \left[\mu_k\tilde{b}_k - \frac{d_k(t)}{B}\right]\]</div>
<p>As one can see, this equation has a steady state if for all i <span class="math notranslate nohighlight">\(\mu_i - \frac{d_i(t)}{B\cdot \tilde{b}_i} = C\)</span> for any constant C:</p>
<p><span class="math">\begin{align}
\frac{d \tilde{b}_i}{dt} &= C\tilde{b}_i - \tilde{b}_i\sum_k C \tilde{b}_k\\
& = \tilde{b}_i\left(C - C\cdot 1) \right) = 0
\end{align}</span></p>
<p>However, in this the total abundance <span class="math notranslate nohighlight">\(B\)</span> would increase indefinitely in time. In particular it holds that <span class="math notranslate nohighlight">\(\frac{dB}{dt} = C\)</span>.</p>
<p>Okay, what can we conclude from that? Most importantly that any abundance <span class="math notranslate nohighlight">\(b_i\)</span> or relative abundance <span class="math notranslate nohighlight">\(\tilde{b}_i\)</span> can be a valid steady state abundance as long as the respective growth and dilution rates are balanced. This means there is an infinite amount of growth rates that yield a steady state (growth rates can not be uniquely identified from abundance alone).</p>
</section>
<section id="Cooperative-tradeoff">
<h2>Cooperative tradeoff<a class="headerlink" href="#Cooperative-tradeoff" title="Permalink to this heading">#</a></h2>
<p>Instead of solving a multi-objective problem MICOM achieves a global optimum by a two step approach called <code class="docutils literal notranslate"><span class="pre">cooperative</span> <span class="pre">tradeoff</span></code>. In simple terms:</p>
<ol class="arabic simple">
<li><p>obtain the maximum community growth rate <span class="math notranslate nohighlight">\(\mu_c^*\)</span> by classic FBA</p></li>
<li><p>choose a tradeoff term <span class="math notranslate nohighlight">\(\alpha\)</span> in <span class="math notranslate nohighlight">\([0, 1]\)</span> and constrain the community growth rate such that <span class="math notranslate nohighlight">\(\mu_c \geq \alpha\cdot\mu_c^*\)</span></p></li>
<li><p>minimize the regularization term <span class="math notranslate nohighlight">\(\sum_i \mu_i^2\)</span> which distributes growth across all individuals in the community</p></li>
</ol>
<p>This method is much faster than alternative methods. Also it results in solution where each individual growth is simultaneously at its maximum rate without diminishing the growth rate of other individuals (thus the term “cooperative”). In consequence, cooperative tradeoff results in an optimum which is conceptually very similar to what OptCom does but with a simpler method. In the absence of constraints MICOM yields growth rate estimates that are correlated with abundance <em>within a single
sample</em>. This is not an assumption made <em>a priori</em> but a consequence of the optimization strategy.</p>
</section>
<section id="Numerical-stability-and-solvers">
<h2>Numerical stability and solvers<a class="headerlink" href="#Numerical-stability-and-solvers" title="Permalink to this heading">#</a></h2>
<p>MICOM support all solver implemented in optlang (GLPK, CPLEX and Gurobi). However, since MICOM models can become quite large (50k - 500k variables) many solvers have numerical issues when solving the corresponding LP and QP problems. For realistic communities modeled on the genus, species or strain level we thus recommend using a solver that implements numerically stable interior point methods, currently only Cplex and Gurobi. Both have academic licenses available. In our experience the simpex
methods in GLPK still work well up to 10,000 variables, but get gradually slower towards 50,000 variables and being too slow to be usable beyond that.</p>
<p>Nevertheless, even the commercial solvers can usaully not control numerical issues in the quadratic interior point methods well. In fact for larger models they always never return an optimal solution. To avoid those numerical issues MICOM employs a crossover strategy where it will perform as many iterations in the quadratic interior point method as possible to get a candidate solution for the individual growth rates <span class="math notranslate nohighlight">\(\mu_i^{ca}\)</span>, followed by solving a linear programming problem to find a
feasible solution close to the candidates:</p>
<div class="math notranslate nohighlight">
\[\begin{split}\begin{align}
\text{maximize } &amp; \mu_c = \sum_i b_i/B \mu^i\\
&amp; s.t. \mu_i \leq \mu_i^{ca}
\end{align}\end{split}\]</div>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[ ]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>
</pre></div>
</div>
</div>
</section>
</section>


                </article>
              

              
              
                <footer class="bd-footer-article">
                  <!-- Previous / next buttons -->
<div class="prev-next-area">
    <a class="left-prev"
       href="index.html"
       title="previous page">
      <i class="fa-solid fa-angle-left"></i>
      <div class="prev-next-info">
        <p class="prev-next-subtitle">previous</p>
        <p class="prev-next-title">Welcome</p>
      </div>
    </a>
    <a class="right-next"
       href="installing.html"
       title="next page">
      <div class="prev-next-info">
        <p class="prev-next-subtitle">next</p>
        <p class="prev-next-title">Installing MICOM</p>
      </div>
      <i class="fa-solid fa-angle-right"></i>
    </a>
</div>
                </footer>
              
            </div>
            
            
              
                <div class="bd-sidebar-secondary bd-toc"><div class="sidebar-secondary-items sidebar-secondary__inner">

  <div class="sidebar-secondary-item">
  <div class="page-toc tocsection onthispage">
    <i class="fa-solid fa-list"></i> Contents
  </div>
  <nav class="bd-toc-nav page-toc">
    <ul class="visible nav section-nav flex-column">
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#Exchanges-and-community-growth-rate">Exchanges and community growth rate</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#Steady-states-in-the-community">Steady states in the community</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#Cooperative-tradeoff">Cooperative tradeoff</a></li>
<li class="toc-h2 nav-item toc-entry"><a class="reference internal nav-link" href="#Numerical-stability-and-solvers">Numerical stability and solvers</a></li>
</ul>
  </nav></div>

</div></div>
              
            
          </div>
          <footer class="bd-footer-content">
            <div class="bd-footer-content__inner">
<div class="bd-footer-content__inner container">
  
  <div class="footer-item">
    
<p class="component-author">
By Christian Diener
</p>

  </div>
  
  <div class="footer-item">
    
  <p class="copyright">
    
      © Copyright 2017-2021, Christian Diener.
      <br/>
    
  </p>

  </div>
  
  <div class="footer-item">
    
  </div>
  
  <div class="footer-item">
    
  </div>
  
</div></div>
          </footer>
        

      </main>
    </div>
  </div>
  
  <!-- Scripts loaded after <body> so the DOM is not blocked -->
  <script src="_static/scripts/bootstrap.js?digest=12da95d707ffb74b382d"></script>
<script src="_static/scripts/pydata-sphinx-theme.js?digest=12da95d707ffb74b382d"></script>

  <footer class="bd-footer">
  </footer>
  </body>
</html>