<!doctype html>
<html class="no-js" lang="en" data-content_root="./">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="viewport" content="width=device-width, initial-scale=1" />
<link rel="index" title="Index" href="genindex.html" /><link rel="search" title="Search" href="search.html" /><link rel="next" title="Writing your own workflows" href="workflows.html" /><link rel="prev" title="Analyzing metabolic interactions" href="interactions.html" />

    <!-- Generated with Sphinx 7.4.7 and Furo 2024.07.18 -->
        <title>Visualizations - version 0.35.0</title>
      <link rel="stylesheet" type="text/css" href="_static/pygments.css?v=a746c00c" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo.css?v=613ab9ff" />
    <link rel="stylesheet" type="text/css" href="_static/graphviz.css?v=fd3f3429" />
    <link rel="stylesheet" type="text/css" href="_static/nbsphinx-code-cells.css" />
    <link rel="stylesheet" type="text/css" href="_static/styles/furo-extensions.css?v=302659d7" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --font-stack: Open Sans, Liberation Sans, sans-serif, Arial;
  --font-stack--monospace: Fira Sans Mono, Inconsolata, monospace, Courier;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-with-moon" viewBox="0 0 24 24">
    <title>Auto light/dark, in light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path style="opacity: 50%" d="M 5.411 14.504 C 5.471 14.504 5.532 14.504 5.591 14.504 C 3.639 16.319 4.383 19.569 6.931 20.352 C 7.693 20.586 8.512 20.551 9.25 20.252 C 8.023 23.207 4.056 23.725 2.11 21.184 C 0.166 18.642 1.702 14.949 4.874 14.536 C 5.051 14.512 5.231 14.5 5.411 14.5 L 5.411 14.504 Z"/>
      <line x1="14.5" y1="3.25" x2="14.5" y2="1.25"/>
      <line x1="14.5" y1="15.85" x2="14.5" y2="17.85"/>
      <line x1="10.044" y1="5.094" x2="8.63" y2="3.68"/>
      <line x1="19" y1="14.05" x2="20.414" y2="15.464"/>
      <line x1="8.2" y1="9.55" x2="6.2" y2="9.55"/>
      <line x1="20.8" y1="9.55" x2="22.8" y2="9.55"/>
      <line x1="10.044" y1="14.006" x2="8.63" y2="15.42"/>
      <line x1="19" y1="5.05" x2="20.414" y2="3.636"/>
      <circle cx="14.5" cy="9.55" r="3.6"/>
    </svg>
  </symbol>
  <symbol id="svg-moon-with-sun" viewBox="0 0 24 24">
    <title>Auto light/dark, in dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round"
      class="icon-custom-derived-from-feather-sun-and-tabler-moon">
      <path d="M 8.282 7.007 C 8.385 7.007 8.494 7.007 8.595 7.007 C 5.18 10.184 6.481 15.869 10.942 17.24 C 12.275 17.648 13.706 17.589 15 17.066 C 12.851 22.236 5.91 23.143 2.505 18.696 C -0.897 14.249 1.791 7.786 7.342 7.063 C 7.652 7.021 7.965 7 8.282 7 L 8.282 7.007 Z"/>
      <line style="opacity: 50%" x1="18" y1="3.705" x2="18" y2="2.5"/>
      <line style="opacity: 50%" x1="18" y1="11.295" x2="18" y2="12.5"/>
      <line style="opacity: 50%" x1="15.316" y1="4.816" x2="14.464" y2="3.964"/>
      <line style="opacity: 50%" x1="20.711" y1="10.212" x2="21.563" y2="11.063"/>
      <line style="opacity: 50%" x1="14.205" y1="7.5" x2="13.001" y2="7.5"/>
      <line style="opacity: 50%" x1="21.795" y1="7.5" x2="23" y2="7.5"/>
      <line style="opacity: 50%" x1="15.316" y1="10.184" x2="14.464" y2="11.036"/>
      <line style="opacity: 50%" x1="20.711" y1="4.789" x2="21.563" y2="3.937"/>
      <circle style="opacity: 50%" cx="18" cy="7.5" r="2.169"/>
    </svg>
  </symbol>
  <symbol id="svg-pencil" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-pencil-code">
      <path d="M4 20h4l10.5 -10.5a2.828 2.828 0 1 0 -4 -4l-10.5 10.5v4" />
      <path d="M13.5 6.5l4 4" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
  <symbol id="svg-eye" viewBox="0 0 24 24">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-eye-code">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M10 12a2 2 0 1 0 4 0a2 2 0 0 0 -4 0" />
      <path
        d="M11.11 17.958c-3.209 -.307 -5.91 -2.293 -8.11 -5.958c2.4 -4 5.4 -6 9 -6c3.6 0 6.6 2 9 6c-.21 .352 -.427 .688 -.647 1.008" />
      <path d="M20 21l2 -2l-2 -2" />
      <path d="M17 17l-2 2l2 2" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>

<a class="skip-to-content muted-link" href="#furo-main-content">Skip to content</a>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="index.html"><div class="brand">version 0.35.0</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
          <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="_static/micom.png" alt="Logo"/>
  </div>
  
  <span class="sidebar-brand-text">version 0.35.0</span>
  
</a><form class="sidebar-search-container" method="get" action="search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="logic.html">Strategies used by MICOM</a></li>
<li class="toctree-l1"><a class="reference internal" href="installing.html">Installing MICOM</a></li>
<li class="toctree-l1"><a class="reference internal" href="high_level.html">MICOM workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="interactions.html">Interactions</a></li>
<li class="toctree-l1 current current-page"><a class="current reference internal" href="#">Visualizations</a></li>
<li class="toctree-l1"><a class="reference internal" href="workflows.html">Writing your own workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="community.html">Single Model API</a></li>
<li class="toctree-l1"><a class="reference internal" href="growth_fluxes.html">Growth rates and fluxes</a></li>
<li class="toctree-l1"><a class="reference internal" href="media.html">Growth media</a></li>
<li class="toctree-l1"><a class="reference internal" href="taxa_knockouts.html">Knockouts</a></li>
<li class="toctree-l1"><a class="reference internal" href="elasticities.html">Intervention studies</a></li>
<li class="toctree-l1"><a class="reference internal" href="tutorials.html">Tutorials and Courses</a></li>
<li class="toctree-l1"><a class="reference internal" href="NEWS.html">Release Notes</a></li>
<li class="toctree-l1 has-children"><a class="reference internal" href="autoapi/index.html">API Reference</a><input class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle navigation of API Reference</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l2 has-children"><a class="reference internal" href="autoapi/micom/index.html">micom</a><input class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle navigation of micom</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/data/index.html">micom.data</a></li>
<li class="toctree-l3 has-children"><a class="reference internal" href="autoapi/micom/interaction/index.html">micom.interaction</a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" role="switch" type="checkbox"/><label for="toctree-checkbox-3"><div class="visually-hidden">Toggle navigation of micom.interaction</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/interaction/focal/index.html">micom.interaction.focal</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/interaction/scores/index.html">micom.interaction.scores</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/interaction/summary/index.html">micom.interaction.summary</a></li>
</ul>
</li>
<li class="toctree-l3 has-children"><a class="reference internal" href="autoapi/micom/viz/index.html">micom.viz</a><input class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" role="switch" type="checkbox"/><label for="toctree-checkbox-4"><div class="visually-hidden">Toggle navigation of micom.viz</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/association/index.html">micom.viz.association</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/core/index.html">micom.viz.core</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/exchanges/index.html">micom.viz.exchanges</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/growth/index.html">micom.viz.growth</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/interactions/index.html">micom.viz.interactions</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/viz/tradeoff/index.html">micom.viz.tradeoff</a></li>
</ul>
</li>
<li class="toctree-l3 has-children"><a class="reference internal" href="autoapi/micom/workflows/index.html">micom.workflows</a><input class="toctree-checkbox" id="toctree-checkbox-5" name="toctree-checkbox-5" role="switch" type="checkbox"/><label for="toctree-checkbox-5"><div class="visually-hidden">Toggle navigation of micom.workflows</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/build/index.html">micom.workflows.build</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/core/index.html">micom.workflows.core</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/db_media/index.html">micom.workflows.db_media</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/grow/index.html">micom.workflows.grow</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/media/index.html">micom.workflows.media</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/results/index.html">micom.workflows.results</a></li>
<li class="toctree-l4"><a class="reference internal" href="autoapi/micom/workflows/tradeoff/index.html">micom.workflows.tradeoff</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/algorithms/index.html">micom.algorithms</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/annotation/index.html">micom.annotation</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/community/index.html">micom.community</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/db/index.html">micom.db</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/deps/index.html">micom.deps</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/duality/index.html">micom.duality</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/elasticity/index.html">micom.elasticity</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/logger/index.html">micom.logger</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/measures/index.html">micom.measures</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/media/index.html">micom.media</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/optcom/index.html">micom.optcom</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/problems/index.html">micom.problems</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/qiime_formats/index.html">micom.qiime_formats</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/solution/index.html">micom.solution</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/stats/index.html">micom.stats</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/taxonomy/index.html">micom.taxonomy</a></li>
<li class="toctree-l3"><a class="reference internal" href="autoapi/micom/util/index.html">micom.util</a></li>
</ul>
</li>
</ul>
</li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          <div class="view-this-page">
  <a class="muted-link" href="_sources/viz.ipynb.txt" title="View this page">
    <svg><use href="#svg-eye"></use></svg>
    <span class="visually-hidden">View this page</span>
  </a>
</div>
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto-light"><use href="#svg-sun-with-moon"></use></svg>
              <svg class="theme-icon-when-auto-dark"><use href="#svg-moon-with-sun"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main" id="furo-main-content">
          <section id="Visualizations">
<h1>Visualizations<a class="headerlink" href="#Visualizations" title="Link to this heading">¶</a></h1>
<p>MICOM provides a a set of visualizations that can be used with the outputs from MICOM workflows. Those visualizations are the same as provided by the <a class="reference external" href="https://github.com/micom-dev/q2-micom">MICOM Qiime 2 plugin</a> but are delivered as single HTML files that bundles interactive graphics and raw data.</p>
<p>To create some more interesting figures here we will use a realistic example data set which is the output of running the MICOM grow workflow on a data set of 10 healthy fecal samples from the <a class="reference external" href="https://www.hmpdacc.org/ihmp/">iHMP cohort</a>. To see the interactive version of a visualization you can click on the provided previews. All visualization contain download buttons to download the raw data used to generate the plot.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[1]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>import micom.data as mmd

results = mmd.test_results()
tradeoff = mmd.test_tradeoff()
</pre></div>
</div>
</div>
<section id="Choosing-a-tradeoff-value">
<h2>Choosing a tradeoff value<a class="headerlink" href="#Choosing-a-tradeoff-value" title="Link to this heading">¶</a></h2>
<p>In the original MICOM publication we chose the tradeoff based on comparisons with <em>in vivo</em> replication rates derived from metagenome data. However, we observed that the highest correlation with replication rates is usually achieved at the largest tradeoff value that allows the majority of the taxa to grow. Thus, we can run cooperative tradeoff with varying tradeoff values and look for the characteristic elbow where the majority of the community can grow. This can be done by using the
<code class="docutils literal notranslate"><span class="pre">plot_tradeoff</span></code> function.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[2]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from micom.viz import plot_tradeoff

pl = plot_tradeoff(tradeoff, filename=&quot;tradeoff.html&quot;)
</pre></div>
</div>
</div>
<p>The returned object is a <code class="docutils literal notranslate"><span class="pre">Visualization</span></code> object that contains the raw data in the data attribute.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>pl
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[3]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
&lt;micom.viz.core.Visualization at 0x7fe492ffce90&gt;
</pre></div></div>
</div>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>pl.data.keys()
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[4]:
</pre></div>
</div>
<div class="output_area docutils container">
<div class="highlight"><pre>
dict_keys([&#39;tradeoff&#39;])
</pre></div></div>
</div>
<p>You could open the visualization in your browser with <code class="docutils literal notranslate"><span class="pre">pl.view()</span></code>. Alternatively you can just open the generated HTML file which would give you something like this:</p>
<p><a class="reference external" href="_static/tradeoff.html"><img alt="tradeoff" src="_images/tradeoff.png" /></a></p>
</section>
<section id="Plotting-growth-rates">
<h2>Plotting growth rates<a class="headerlink" href="#Plotting-growth-rates" title="Link to this heading">¶</a></h2>
<p>The first thing we may want to investigate are the growth rates predicted by MICOM. This can be done with the <code class="docutils literal notranslate"><span class="pre">plot_growth</span></code> function.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[5]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from micom.viz import plot_growth

pl = plot_growth(results, filename=&quot;growth_rates.html&quot;)
</pre></div>
</div>
</div>
<p>Which will give you the following:</p>
<p><a class="reference external" href="_static/growth_rates.html"><img alt="growth rates" src="_images/growth_rates.png" /></a></p>
</section>
<section id="Plotting-consumed-metabolites">
<h2>Plotting consumed metabolites<a class="headerlink" href="#Plotting-consumed-metabolites" title="Link to this heading">¶</a></h2>
<p>To get an overview which metabolites are consumed by the entire microbiota we can use the <code class="docutils literal notranslate"><span class="pre">plot_exchanges_per_sample</span></code> function.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[6]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from micom.viz import plot_exchanges_per_sample

pl = plot_exchanges_per_sample(results, filename=&quot;consumption.html&quot;)
</pre></div>
</div>
</div>
<p>This will give you a heatmap showing all consumed components. Unless specified otherwise in the function arguments samples will be clustered so that samples with similar consumption profiles will be close.</p>
<p><a class="reference external" href="_static/consumption.html"><img alt="consumption" src="_images/consumption.png" /></a></p>
</section>
<section id="Plotting-growth-niches">
<h2>Plotting growth niches<a class="headerlink" href="#Plotting-growth-niches" title="Link to this heading">¶</a></h2>
<p>What is consumed globally may be interesting but we may want to know even more how the available growth niches are occupied by the taxa in the sample. This can be done with <code class="docutils literal notranslate"><span class="pre">plot_exchanges_per_taxon</span></code> which will embed the import fluxes for each taxon into two dimension using TSNE and plot the niche occupation map. Here taxa that overlap compete for similar sets of resources. The center of the map denotes the most competitive niche whereas the outskirts of the map denote more specialized
consumption preferences.</p>
<div class="nbinput docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[7]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from micom.viz import plot_exchanges_per_taxon

pl = plot_exchanges_per_taxon(results, filename=&quot;niche.html&quot;)
</pre></div>
</div>
</div>
<div class="nboutput nblast docutils container">
<div class="prompt empty docutils container">
</div>
<div class="output_area rendered_html docutils container">
<pre style="white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace"><span style="color: #7fbfbf; text-decoration-color: #7fbfbf">[12:54:10] </span><span style="color: #800000; text-decoration-color: #800000">WARNING </span> Not enough samples. Adjusting T-SNE perplexity to <span style="color: #008080; text-decoration-color: #008080; font-weight: bold">5</span>.                           <a href="file:///home/cdiener/code/micom/micom/viz/exchanges.py" target="_blank"><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">exchanges.py</span></a><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">:</span><a href="file:///home/cdiener/code/micom/micom/viz/exchanges.py#127" target="_blank"><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">127</span></a>
</pre></div>
</div>
<p>This will give you the following:</p>
<p><a class="reference external" href="_static/niche.html"><img alt="niche" src="_images/niche.png" /></a></p>
</section>
<section id="Investigating-associations-with-a-phenotype">
<h2>Investigating associations with a phenotype<a class="headerlink" href="#Investigating-associations-with-a-phenotype" title="Link to this heading">¶</a></h2>
<p>Finally we may want to which fluxes relate to an observed phenotype. This can be done with the <code class="docutils literal notranslate"><span class="pre">plot_association</span></code> function which will:</p>
<ol class="arabic simple">
<li><p>calculate overall production or consumption fluxes (total metabolite amount produced by the microbiota)</p></li>
<li><p>run non-parametric tests for each metabolite against the phenotype</p></li>
<li><p>control the flase discovery rate and report significantly associated metabolite fluxes</p></li>
<li><p>log-transform and standardize production fluxes</p></li>
<li><p>train a LASSO regression (continuous response) or LASSO logistic regression (binary response)</p></li>
<li><p>present the overall performance of the fluxes inpredicting the phenotype</p></li>
</ol>
<p>So you will get data on local (metabolite) and global (all fluxes) associations.</p>
<p>To illustrate this we will create a mock phenotype that is correlated with propionate production. We will allow very high q values here. In a real analysis the default of 0.05 is more appropriate,</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[8]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from micom.viz import plot_association
from micom.measures import production_rates
import numpy as np
import pandas as pd

prod = production_rates(results)
propionate = prod.loc[prod.metabolite == &quot;ppa[e]&quot;, &quot;flux&quot;]
propionate.index = prod.sample_id.unique()
high_propionate = propionate &gt; np.median(propionate)

pl = plot_association(
    results,
    phenotype=high_propionate,
    variable_type=&quot;binary&quot;,
    filename=&quot;association.html&quot;,
    fdr_threshold=0.5,
)
</pre></div>
</div>
</div>
<p>The output will look something like the following.</p>
<p><a class="reference external" href="_static/association.html"><img alt="image of the associations" src="_images/association.png" /></a></p>
<p>So we see we recovered propionate in the analysis but we would need larger sample sizes here.</p>
</section>
<section id="Plotting-interactions">
<h2>Plotting interactions<a class="headerlink" href="#Plotting-interactions" title="Link to this heading">¶</a></h2>
<p>We provide support to plot focal interactions for a taxon of interest or the Metabolic Exchange Score (MES). For instance, let’s start by plotting the interactions for Akkermansia.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[9]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from micom.viz import plot_focal_interactions

pl = plot_focal_interactions(results, taxon=&quot;s__Akkermansia_muciniphila&quot;)
</pre></div>
</div>
</div>
<p>You can choose many types of interaction fluxes by passing the <code class="docutils literal notranslate"><span class="pre">kind</span></code> argument. Currently supported options are: - “flux”: the raw flux - “mass”: the mass flux - “C”: the carbon flux - “N”: the nitrogen flux</p>
<p>This will give you a larger overview that looks like this:</p>
<p><a class="reference external" href="_static/focal.html"><img alt="focal examples" src="_images/focal.png" /></a></p>
<p>Alternatively you can also visualize the Metabolic Exchange Scores. This can be done across different groups as well. To illustrate this let’s do this with some random groups.</p>
<div class="nbinput nblast docutils container">
<div class="prompt highlight-none notranslate"><div class="highlight"><pre><span></span>[12]:
</pre></div>
</div>
<div class="input_area highlight-ipython3 notranslate"><div class="highlight"><pre><span></span>from micom.viz import plot_mes

groups = pd.Series(
    5 * [&quot;a&quot;] + 5 * [&quot;b&quot;],
    index=results.growth_rates.sample_id.unique(),
    name=&quot;random&quot;
)

pl = plot_mes(results, groups=groups, filename=&quot;mes.html&quot;)
</pre></div>
</div>
</div>
<p>Here you see the MES for each metabolite and also a global overview across the groups.</p>
<p><a class="reference external" href="_static/mes.html"><img alt="MES example" src="_images/mes.png" /></a></p>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          <a class="next-page" href="workflows.html">
              <div class="page-info">
                <div class="context">
                  <span>Next</span>
                </div>
                <div class="title">Writing your own workflows</div>
              </div>
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
            </a>
          <a class="prev-page" href="interactions.html">
              <svg class="furo-related-icon"><use href="#svg-arrow-right"></use></svg>
              <div class="page-info">
                <div class="context">
                  <span>Previous</span>
                </div>
                
                <div class="title">Analyzing metabolic interactions</div>
                
              </div>
            </a>
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2017-2024, Christian Diener
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#">Visualizations</a><ul>
<li><a class="reference internal" href="#Choosing-a-tradeoff-value">Choosing a tradeoff value</a></li>
<li><a class="reference internal" href="#Plotting-growth-rates">Plotting growth rates</a></li>
<li><a class="reference internal" href="#Plotting-consumed-metabolites">Plotting consumed metabolites</a></li>
<li><a class="reference internal" href="#Plotting-growth-niches">Plotting growth niches</a></li>
<li><a class="reference internal" href="#Investigating-associations-with-a-phenotype">Investigating associations with a phenotype</a></li>
<li><a class="reference internal" href="#Plotting-interactions">Plotting interactions</a></li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script src="_static/documentation_options.js?v=ca1b380b"></script>
    <script src="_static/doctools.js?v=9a2dae69"></script>
    <script src="_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="_static/scripts/furo.js?v=5fa4622c"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script>window.MathJax = {"tex": {"inlineMath": [["$", "$"], ["\\(", "\\)"]], "processEscapes": true}, "options": {"ignoreHtmlClass": "tex2jax_ignore|mathjax_ignore|document", "processHtmlClass": "tex2jax_process|mathjax_process|math|output_area"}}</script>
    <script defer="defer" src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
    </body>
</html>