<!doctype html>
<html class="no-js" lang="en">
  <head><meta charset="utf-8"/>
    <meta name="viewport" content="width=device-width,initial-scale=1"/>
    <meta name="color-scheme" content="light dark"><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />
<link rel="index" title="Index" href="../../../genindex.html" /><link rel="search" title="Search" href="../../../search.html" />

    <!-- Generated with Sphinx 6.1.3 and Furo 2023.03.27 -->
        <title>micom.workflows - version 0.32.4</title>
      <link rel="stylesheet" type="text/css" href="../../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/furo.css?digest=fad236701ea90a88636c2a8c73b44ae642ed2a53" />
    <link rel="stylesheet" type="text/css" href="../../../_static/graphviz.css" />
    <link rel="stylesheet" type="text/css" href="../../../_static/styles/furo-extensions.css?digest=30d1aed668e5c3a91c3e3bf6a60b675221979f0e" />
    
    


<style>
  body {
    --color-code-background: #f8f8f8;
  --color-code-foreground: black;
  --font-stack: Open Sans, Liberation Sans, sans-serif, Arial;
  --font-stack--monospace: Fira Sans Mono, Inconsolata, monospace, Courier;
  
  }
  @media not print {
    body[data-theme="dark"] {
      --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
    }
    @media (prefers-color-scheme: dark) {
      body:not([data-theme="light"]) {
        --color-code-background: #202020;
  --color-code-foreground: #d0d0d0;
  
      }
    }
  }
</style></head>
  <body>
    
    <script>
      document.body.dataset.theme = localStorage.getItem("theme") || "auto";
    </script>
    

<svg xmlns="http://www.w3.org/2000/svg" style="display: none;">
  <symbol id="svg-toc" viewBox="0 0 24 24">
    <title>Contents</title>
    <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 1024 1024">
      <path d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 0 0 0 13.8z"/>
    </svg>
  </symbol>
  <symbol id="svg-menu" viewBox="0 0 24 24">
    <title>Menu</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-menu">
      <line x1="3" y1="12" x2="21" y2="12"></line>
      <line x1="3" y1="6" x2="21" y2="6"></line>
      <line x1="3" y1="18" x2="21" y2="18"></line>
    </svg>
  </symbol>
  <symbol id="svg-arrow-right" viewBox="0 0 24 24">
    <title>Expand</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather-chevron-right">
      <polyline points="9 18 15 12 9 6"></polyline>
    </svg>
  </symbol>
  <symbol id="svg-sun" viewBox="0 0 24 24">
    <title>Light mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="feather-sun">
      <circle cx="12" cy="12" r="5"></circle>
      <line x1="12" y1="1" x2="12" y2="3"></line>
      <line x1="12" y1="21" x2="12" y2="23"></line>
      <line x1="4.22" y1="4.22" x2="5.64" y2="5.64"></line>
      <line x1="18.36" y1="18.36" x2="19.78" y2="19.78"></line>
      <line x1="1" y1="12" x2="3" y2="12"></line>
      <line x1="21" y1="12" x2="23" y2="12"></line>
      <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
      <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
    </svg>
  </symbol>
  <symbol id="svg-moon" viewBox="0 0 24 24">
    <title>Dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-moon">
      <path stroke="none" d="M0 0h24v24H0z" fill="none" />
      <path d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z" />
    </svg>
  </symbol>
  <symbol id="svg-sun-half" viewBox="0 0 24 24">
    <title>Auto light/dark mode</title>
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor"
      stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round" class="icon-tabler-shadow">
      <path stroke="none" d="M0 0h24v24H0z" fill="none"/>
      <circle cx="12" cy="12" r="9" />
      <path d="M13 12h5" />
      <path d="M13 15h4" />
      <path d="M13 18h1" />
      <path d="M13 9h4" />
      <path d="M13 6h1" />
    </svg>
  </symbol>
</svg>

<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
<label class="overlay sidebar-overlay" for="__navigation">
  <div class="visually-hidden">Hide navigation sidebar</div>
</label>
<label class="overlay toc-overlay" for="__toc">
  <div class="visually-hidden">Hide table of contents sidebar</div>
</label>



<div class="page">
  <header class="mobile-header">
    <div class="header-left">
      <label class="nav-overlay-icon" for="__navigation">
        <div class="visually-hidden">Toggle site navigation sidebar</div>
        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
      </label>
    </div>
    <div class="header-center">
      <a href="../../../index.html"><div class="brand">version 0.32.4</div></a>
    </div>
    <div class="header-right">
      <div class="theme-toggle-container theme-toggle-header">
        <button class="theme-toggle">
          <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
          <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
          <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
          <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
        </button>
      </div>
      <label class="toc-overlay-icon toc-header-icon" for="__toc">
        <div class="visually-hidden">Toggle table of contents sidebar</div>
        <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
      </label>
    </div>
  </header>
  <aside class="sidebar-drawer">
    <div class="sidebar-container">
      
      <div class="sidebar-sticky"><a class="sidebar-brand" href="../../../index.html">
  
  <div class="sidebar-logo-container">
    <img class="sidebar-logo" src="../../../_static/micom.png" alt="Logo"/>
  </div>
  
  <span class="sidebar-brand-text">version 0.32.4</span>
  
</a><form class="sidebar-search-container" method="get" action="../../../search.html" role="search">
  <input class="sidebar-search" placeholder="Search" name="q" aria-label="Search">
  <input type="hidden" name="check_keywords" value="yes">
  <input type="hidden" name="area" value="default">
</form>
<div id="searchbox"></div><div class="sidebar-scroll"><div class="sidebar-tree">
  <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../logic.html">Methods used by MICOM</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../installing.html">Installing MICOM</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../high_level.html">MICOM workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../viz.html">Visualizations</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../workflows.html">Writing your own workflows</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../community.html">Single Model API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../growth_fluxes.html">Growth rates and fluxes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../media.html">Growth media</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../taxa_knockouts.html">Knockouts</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../elasticities.html">Intervention studies</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials.html">Tutorials and Courses</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../NEWS.html">Release Notes</a></li>
<li class="toctree-l1 current has-children"><a class="reference internal" href="../../index.html">API Reference</a><input checked="" class="toctree-checkbox" id="toctree-checkbox-1" name="toctree-checkbox-1" role="switch" type="checkbox"/><label for="toctree-checkbox-1"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul class="current">
<li class="toctree-l2 current has-children"><a class="reference internal" href="../index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom</span></code></a><input checked="" class="toctree-checkbox" id="toctree-checkbox-2" name="toctree-checkbox-2" role="switch" type="checkbox"/><label for="toctree-checkbox-2"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="../data/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.data</span></code></a></li>
<li class="toctree-l3 has-children"><a class="reference internal" href="../viz/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz</span></code></a><input class="toctree-checkbox" id="toctree-checkbox-3" name="toctree-checkbox-3" role="switch" type="checkbox"/><label for="toctree-checkbox-3"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l4"><a class="reference internal" href="../viz/core/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz.core</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../viz/exchanges/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz.exchanges</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../viz/growth/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz.growth</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../viz/prediction/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz.prediction</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../viz/tradeoff/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.viz.tradeoff</span></code></a></li>
</ul>
</li>
<li class="toctree-l3 current has-children current-page"><a class="current reference internal" href="#"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows</span></code></a><input checked="" class="toctree-checkbox" id="toctree-checkbox-4" name="toctree-checkbox-4" role="switch" type="checkbox"/><label for="toctree-checkbox-4"><div class="visually-hidden">Toggle child pages in navigation</div><i class="icon"><svg><use href="#svg-arrow-right"></use></svg></i></label><ul>
<li class="toctree-l4"><a class="reference internal" href="build/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.build</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="core/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.core</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="db_media/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.db_media</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="grow/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.grow</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="media/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.media</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="tradeoff/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.tradeoff</span></code></a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="../algorithms/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.algorithms</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../annotation/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.annotation</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../community/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.community</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../db/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.db</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../deps/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.deps</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../duality/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.duality</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../elasticity/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.elasticity</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../logger/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.logger</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../measures/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.measures</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../media/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.media</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../optcom/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.optcom</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../problems/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.problems</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../qiime_formats/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.qiime_formats</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../solution/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.solution</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../stats/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.stats</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../taxonomy/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.taxonomy</span></code></a></li>
<li class="toctree-l3"><a class="reference internal" href="../util/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.util</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>

</div>
</div>

      </div>
      
    </div>
  </aside>
  <div class="main">
    <div class="content">
      <div class="article-container">
        <a href="#" class="back-to-top muted-link">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12z"></path>
          </svg>
          <span>Back to top</span>
        </a>
        <div class="content-icon-container">
          
<div class="theme-toggle-container theme-toggle-content">
            <button class="theme-toggle">
              <div class="visually-hidden">Toggle Light / Dark / Auto color theme</div>
              <svg class="theme-icon-when-auto"><use href="#svg-sun-half"></use></svg>
              <svg class="theme-icon-when-dark"><use href="#svg-moon"></use></svg>
              <svg class="theme-icon-when-light"><use href="#svg-sun"></use></svg>
            </button>
          </div>
          <label class="toc-overlay-icon toc-content-icon" for="__toc">
            <div class="visually-hidden">Toggle table of contents sidebar</div>
            <i class="icon"><svg><use href="#svg-toc"></use></svg></i>
          </label>
        </div>
        <article role="main">
          <section id="module-micom.workflows">
<span id="micom-workflows"></span><h1><a class="reference internal" href="#module-micom.workflows" title="micom.workflows"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows</span></code></a><a class="headerlink" href="#module-micom.workflows" title="Permalink to this heading">#</a></h1>
<p>Init file for MICOM workflows.</p>
<section id="submodules">
<h2>Submodules<a class="headerlink" href="#submodules" title="Permalink to this heading">#</a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="build/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.build</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="core/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.core</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="db_media/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.db_media</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="grow/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.grow</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="media/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.media</span></code></a></li>
<li class="toctree-l1"><a class="reference internal" href="tradeoff/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows.tradeoff</span></code></a></li>
</ul>
</div>
</section>
<section id="package-contents">
<h2>Package Contents<a class="headerlink" href="#package-contents" title="Permalink to this heading">#</a></h2>
<section id="functions">
<h3>Functions<a class="headerlink" href="#functions" title="Permalink to this heading">#</a></h3>
<div class="table-wrapper autosummary longtable docutils container">
<table class="autosummary longtable docutils align-default">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#micom.workflows.workflow" title="micom.workflows.workflow"><code class="xref py py-obj docutils literal notranslate"><span class="pre">workflow</span></code></a>(func, args[, threads, description, progress])</p></td>
<td><p>Run analyses for several samples in parallel.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#micom.workflows.save_results" title="micom.workflows.save_results"><code class="xref py py-obj docutils literal notranslate"><span class="pre">save_results</span></code></a>(results, path)</p></td>
<td><p>Save growth results to a file.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#micom.workflows.load_results" title="micom.workflows.load_results"><code class="xref py py-obj docutils literal notranslate"><span class="pre">load_results</span></code></a>(path)</p></td>
<td><p>Load growth results from a file.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#micom.workflows.build" title="micom.workflows.build"><code class="xref py py-obj docutils literal notranslate"><span class="pre">build</span></code></a></p></td>
<td><p>Worflow to build models for several samples.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#micom.workflows.build_database" title="micom.workflows.build_database"><code class="xref py py-obj docutils literal notranslate"><span class="pre">build_database</span></code></a>(manifest, out_path[, rank, threads, ...])</p></td>
<td><p>Create a model database from a set of SBML files.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#micom.workflows.grow" title="micom.workflows.grow"><code class="xref py py-obj docutils literal notranslate"><span class="pre">grow</span></code></a></p></td>
<td><p>Performs growth and exchange analysis for several models.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="tradeoff/index.html#module-micom.workflows.tradeoff" title="micom.workflows.tradeoff"><code class="xref py py-obj docutils literal notranslate"><span class="pre">tradeoff</span></code></a>(manifest, model_folder, medium[, tradeoffs, ...])</p></td>
<td><p>Run growth rate predictions for varying tradeoff values.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#micom.workflows.fix_medium" title="micom.workflows.fix_medium"><code class="xref py py-obj docutils literal notranslate"><span class="pre">fix_medium</span></code></a>(manifest, model_folder, medium[, ...])</p></td>
<td><p>Augment a growth medium so all community members can grow in it.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#micom.workflows.minimal_media" title="micom.workflows.minimal_media"><code class="xref py py-obj docutils literal notranslate"><span class="pre">minimal_media</span></code></a>(manifest, model_folder[, summarize, ...])</p></td>
<td><p>Calculate the minimal medium for a set of community models.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#micom.workflows.check_db_medium" title="micom.workflows.check_db_medium"><code class="xref py py-obj docutils literal notranslate"><span class="pre">check_db_medium</span></code></a>(model_db, medium[, threads])</p></td>
<td><p>Complete a growth medium for all models in a database.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#micom.workflows.complete_db_medium" title="micom.workflows.complete_db_medium"><code class="xref py py-obj docutils literal notranslate"><span class="pre">complete_db_medium</span></code></a>(model_db, medium[, growth, ...])</p></td>
<td><p>Complete a growth medium for all models in a database.</p></td>
</tr>
</tbody>
</table>
</div>
</section>
<section id="attributes">
<h3>Attributes<a class="headerlink" href="#attributes" title="Permalink to this heading">#</a></h3>
<div class="table-wrapper autosummary longtable docutils container">
<table class="autosummary longtable docutils align-default">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#micom.workflows.GrowthResults" title="micom.workflows.GrowthResults"><code class="xref py py-obj docutils literal notranslate"><span class="pre">GrowthResults</span></code></a></p></td>
<td><p></p></td>
</tr>
</tbody>
</table>
</div>
<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.workflow">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">workflow</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">func</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">args</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">4</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">description</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">progress</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../../../_modules/micom/workflows/core.html#workflow"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#micom.workflows.workflow" title="Permalink to this definition">#</a></dt>
<dd><p>Run analyses for several samples in parallel.</p>
<p>This will analyze several samples in parallel. Includes a workaround for
optlang memory leak.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>func</strong> (<em>function</em>) – A function that takes a single argument (can be any object) and
that performs your analysis for a single sample.</p></li>
<li><p><strong>args</strong> (<em>array-like object</em>) – An array-like object (list, tuple, numpy array, pandas Series, etc.)
that contains the arguments for each sample.</p></li>
<li><p><strong>threads</strong> (<em>positive int</em>) – How many samples to analyze in parallel at once.</p></li>
<li><p><strong>description</strong> (<em>str</em>) – The dewscription shown in front of the progress bar.</p></li>
<li><p><strong>progress</strong> (<em>bool</em>) – Whether to show a progress bar.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py data">
<dt class="sig sig-object py" id="micom.workflows.GrowthResults">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">GrowthResults</span></span><a class="reference internal" href="../../../_modules/micom/workflows/core.html#GrowthResults"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#micom.workflows.GrowthResults" title="Permalink to this definition">#</a></dt>
<dd></dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.save_results">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">save_results</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">results</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">path</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../../../_modules/micom/workflows/core.html#save_results"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#micom.workflows.save_results" title="Permalink to this definition">#</a></dt>
<dd><p>Save growth results to a file.</p>
<p>This will write all tables as CSV into a single ZIP file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>results</strong> (<em>GrowthResults</em>) – The results as returned from <cite>grow</cite>.</p></li>
<li><p><strong>path</strong> (<em>str</em>) – A filepath for the generated file. Should end in <cite>.zip</cite>.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.load_results">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">load_results</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">path</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../../../_modules/micom/workflows/core.html#load_results"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#micom.workflows.load_results" title="Permalink to this definition">#</a></dt>
<dd><p>Load growth results from a file.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><p><strong>path</strong> (<em>str</em>) – Path to saved <cite>GrowthResults</cite>.</p>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The saved GrowthResults.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>GrowthResults</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.build">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">build</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">taxonomy</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_db</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_folder</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">cutoff</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.0001</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">solver</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#micom.workflows.build" title="Permalink to this definition">#</a></dt>
<dd><p>Build a series of community models.</p>
<p>This is a best-practice implementation of building community models
for several samples in parallel.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>taxonomy</strong> (<em>pandas.DataFrame</em>) – The taxonomy used for building the model. Must have at least the
columns “id” and “sample_id”. This must also
contain at least a column with the same name as the rank used in
the model database. Thus, for a genus-level database you will need
a column <cite>genus</cite>. Additional taxa ranks can also be specified and
will be used to be more stringent in taxa matching.
Finally, the taxonomy should contain a column <cite>abundance</cite>. It will
be used to quantify each individual in the community. If absent,
MICOM will assume all individuals are present in the same amount.</p></li>
<li><p><strong>model_db</strong> (<em>str</em>) – A pre-built model database. If ending in <cite>.qza</cite> must be a Qiime 2
artifact of type <cite>MetabolicModels[JSON]</cite>. Can also be a folder,
zip (must end in <cite>.zip</cite>) file or None if the taxonomy contains a
column <cite>file</cite>.</p></li>
<li><p><strong>out_folder</strong> (<em>str</em>) – The built models and a manifest file will be written to this
folder. Will continue</p></li>
<li><p><strong>cutoff</strong> (<em>float in</em><em> [</em><em>0.0</em><em>, </em><em>1.0</em><em>]</em>) – Abundance cutoff. Taxa with a relative abundance smaller than this
will not be included in the model.</p></li>
<li><p><strong>threads</strong> (<em>int &gt;=1</em>) – The number of parallel workers to use when building models. As a
rule of thumb you will need around 1GB of RAM for each thread.</p></li>
<li><p><strong>solver</strong> (<em>str</em>) – Name of the solver used for the linear and quadratic problems.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The manifest for the built models. Contains taxa abundances,
build metrics and file basenames.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas.DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.build_database">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">build_database</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">manifest</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">out_path</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rank</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'genus'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">compress</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">compresslevel</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">6</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">progress</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../../../_modules/micom/workflows/build.html#build_database"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#micom.workflows.build_database" title="Permalink to this definition">#</a></dt>
<dd><p>Create a model database from a set of SBML files.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>A manifest for the joined models will also be written to the output folder
as “manifest.csv”. This may contain NA entries for additional columns
that had different values within the summarized models.</p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>manifest</strong> (<em>pandas.DataFrame</em>) – A manifest of SBML files containing their filepath as well as taxonomy.
Must contain the columns “file”, “kingdom”, “phylum”, “class”,
“order”, “family”, “genus”, and “species”. May contain additional
columns.</p></li>
<li><p><strong>out_path</strong> (<em>str</em>) – The directory or zip file where the joined models will be written.</p></li>
<li><p><strong>threads</strong> (<em>int &gt;=1</em>) – The number of parallel workers to use when building models. As a
rule of thumb you will need around 1GB of RAM for each thread.</p></li>
<li><p><strong>compress</strong> (<em>str</em><em> (</em><em>default None</em><em>)</em>) – Compression method to use. Must be “zlib”, “bz2”, “lzma” or None.
This parameter is ignored if out_path does not end with “.zip”.</p></li>
<li><p><strong>compresslevel</strong> (<em>int</em><em> [</em><em>1-9</em><em>] </em><em>(</em><em>default: 6</em><em>)</em>) – Level of compression. Only used if compress is not None.
This parameter is ignored if out_path does not end with “.zip”.</p></li>
<li><p><strong>progress</strong> (<em>bool</em>) – Whether to show a progress bar.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The manifest of the joined models. Will still contain information
from the original metadata.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.grow">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">grow</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">manifest</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_folder</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">medium</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tradeoff</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weights</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">strategy</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">'minimal</span> <span class="pre">imports'</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rtol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">presolve</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#micom.workflows.grow" title="Permalink to this definition">#</a></dt>
<dd><p>Simulate growth for a set of community models.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The strategy <cite>mimimal imports</cite> can become unstable for common carbon sources since
it will add in infeasible imports that are very small but import some high-C
molecules. If you use it check that only components from your medium have been used
and molecules that should be essential are indeed consumed.</p>
</div>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>manifest</strong> (<em>pandas.DataFrame</em>) – The manifest as returned by the <cite>build</cite> workflow.</p></li>
<li><p><strong>model_folder</strong> (<em>str</em>) – The folder in which to find the files mentioned in the manifest.</p></li>
<li><p><strong>medium</strong> (<em>pandas.DataFrame</em>) – A growth medium. Must have columns “reaction” and “flux” denoting
exchange reactions and their respective maximum flux.</p></li>
<li><p><strong>tradeoff</strong> (<em>float in</em><em> (</em><em>0.0</em><em>, </em><em>1.0</em><em>]</em>) – A tradeoff value. Can be chosen by running the <cite>tradeoff</cite> workflow or
by experince. Tradeoff values of 0.5 for metagenomcis data and 0.3 for
16S data seem to work well.</p></li>
<li><p><strong>threads</strong> (<em>int &gt;=1</em>) – The number of parallel workers to use when building models. As a
rule of thumb you will need around 1GB of RAM for each thread.</p></li>
<li><p><strong>strategy</strong> (<em>str</em>) – Computational strategy used to reduce the flux space. Default “pFBA” uses
parsimonious FBA, “minimal imports” used the solution with the smallest
total mass import from the environment, and “none” returns an arbitrary
feasible flux distribution.</p></li>
<li><p><strong>weights</strong> (<em>str</em>) – Only used during the calculaton of the minimal import rates.
Will scale the fluxes by a weight factor. Can either be “mass” which will
scale by molecular mass, a single element which will scale by
the elemental content (for instance “C” to scale by carbon content).
If None every metabolite will receive the same weight.
Will be ignored if <cite>minimize_components</cite> is True.</p></li>
<li><p><strong>atol</strong> (<em>float</em>) – Absolute tolerance for the growth rates. If None will use the solver tolerance.</p></li>
<li><p><strong>rtol</strong> (<em>float</em>) – Relative tolerqance for the growth rates. If None will use the solver tolerance.</p></li>
<li><p><strong>presolve</strong> (<em>bool</em>) – Whether to use the presolver/scaling. Can improve numerical accuracy in some
cases.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A named tuple containing the growth rates and exchange fluxes for all
samples/models.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>GrowthResults</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.tradeoff">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">tradeoff</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">manifest</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_folder</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">medium</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">tradeoffs</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">np.arange(0.1,</span> <span class="pre">1.0</span> <span class="pre">+</span> <span class="pre">1e-06,</span> <span class="pre">0.1)</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">atol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">rtol</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">presolve</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../../../_modules/micom/workflows/tradeoff.html#tradeoff"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#micom.workflows.tradeoff" title="Permalink to this definition">#</a></dt>
<dd><p>Run growth rate predictions for varying tradeoff values.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>manifest</strong> (<em>pandas.DataFrame</em>) – The manifest as returned by the <cite>build</cite> workflow.</p></li>
<li><p><strong>model_folder</strong> (<em>str</em>) – The folder in which to find the files mentioned in the manifest.</p></li>
<li><p><strong>medium</strong> (<em>pandas.DataFrame</em>) – A growth medium. Must have columns “reaction” and “flux” denoting
exchnage reactions and their respective maximum flux.</p></li>
<li><p><strong>tradeoffs</strong> (<em>array</em><em> of </em><em>floats in</em><em> (</em><em>0.0</em><em>, </em><em>1.0</em><em>]</em>) – An array of tradeoff vaues to be tested. One simulation without
a tradeoff (no cooperative tradeoff) will always be run additionally
and will have a tradeoff of “NaN”.</p></li>
<li><p><strong>threads</strong> (<em>int &gt;=1</em>) – The number of parallel workers to use when building models. As a
rule of thumb you will need around 1GB of RAM for each thread.</p></li>
<li><p><strong>atol</strong> (<em>float</em>) – Absolute tolerance for the growth rates. If None will use the solver tolerance.</p></li>
<li><p><strong>rtol</strong> (<em>float</em>) – Relative tolerqance for the growth rates. If None will use the solver tolerance.</p></li>
<li><p><strong>presolve</strong> (<em>bool</em>) – Whether to use the presolver/scaling. Can improve numerical accuracy in some
cases.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>The predicted growth rates.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas.DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.fix_medium">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">fix_medium</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">manifest</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_folder</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">medium</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">community_growth</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_growth</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.001</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_import</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minimize_components</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">summarize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weights</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../../../_modules/micom/workflows/media.html#fix_medium"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#micom.workflows.fix_medium" title="Permalink to this definition">#</a></dt>
<dd><p>Augment a growth medium so all community members can grow in it.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>manifest</strong> (<em>pandas.DataFrame</em>) – The manifest as returned by the <cite>build</cite> workflow.</p></li>
<li><p><strong>model_folder</strong> (<em>str</em>) – The folder in which to find the files mentioned in the manifest.</p></li>
<li><p><strong>medium</strong> (<em>pandas.Series</em><em> or </em><em>pandas.DataFrame</em>) – A growth medium with exchange reaction IDs as index and positive
import fluxes as values. If a DataFrame needs columns <cite>flux</cite> and
<cite>reaction</cite>.</p></li>
<li><p><strong>community_growth</strong> (<em>positive float</em>) – The minimum community-wide growth rate that has to be achieved on the created
medium.</p></li>
<li><p><strong>min_growth</strong> (<em>positive float</em>) – The minimum biomass production required for growth.</p></li>
<li><p><strong>max_import</strong> (<em>positive float</em>) – The maximum import rate for added imports.</p></li>
<li><p><strong>minimize_components</strong> (<em>boolean</em>) – Whether to minimize the number of media components rather than the
total flux.</p></li>
<li><p><strong>summarize</strong> (<em>boolean</em>) – Whether to summarize the medium across all samples. If False will
return a medium for each sample.</p></li>
<li><p><strong>weights</strong> (<em>str</em>) – Will scale the fluxes by a weight factor. Can either be “mass” which will
scale by molecular mass, a single element which will scale by
the elemental content (for instance “C” to scale by carbon content).
If None every metabolite will receive the same weight.
Will be ignored if <cite>minimize_components</cite> is True.</p></li>
<li><p><strong>threads</strong> (<em>int</em>) – The number of processes to use.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>A new growth medium with the smallest amount of augmentations such
that all members of the community can grow in it.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pandas.DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.minimal_media">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">minimal_media</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">manifest</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">model_folder</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">summarize</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">True</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">min_growth</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../../../_modules/micom/workflows/media.html#minimal_media"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#micom.workflows.minimal_media" title="Permalink to this definition">#</a></dt>
<dd><p>Calculate the minimal medium for a set of community models.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.check_db_medium">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">check_db_medium</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model_db</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">medium</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../../../_modules/micom/workflows/db_media.html#check_db_medium"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#micom.workflows.check_db_medium" title="Permalink to this definition">#</a></dt>
<dd><p>Complete a growth medium for all models in a database.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>model_db</strong> (<em>str</em>) – A pre-built model database. If ending in <cite>.qza</cite> must be a Qiime 2
artifact of type <cite>MetabolicModels[JSON]</cite>. Can also be a folder,
zip (must end in <cite>.zip</cite>) file or None if the taxonomy contains a
column <cite>file</cite>.</p></li>
<li><p><strong>medium</strong> (<em>pd.DataFrame</em>) – A growth medium. Must have columns “reaction” and “flux” denoting
exchange reactions and their respective maximum flux. Can not be sample
specific.</p></li>
<li><p><strong>threads</strong> (<em>int &gt;=1</em>) – The number of parallel workers to use when building models. As a
rule of thumb you will need around 1GB of RAM for each thread.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Returns an annotated manifest file with a column <cite>can_grow</cite> that tells you
whether the model can grow on the (fixed) medium, and a column <cite>growth_rate</cite>
that gives the growth rate.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>pd.DataFrame</p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="micom.workflows.complete_db_medium">
<span class="sig-prename descclassname"><span class="pre">micom.workflows.</span></span><span class="sig-name descname"><span class="pre">complete_db_medium</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">model_db</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">medium</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">growth</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">0.001</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">max_added_import</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">minimize_components</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">False</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">weights</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">None</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">threads</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">1</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">strict</span></span><span class="o"><span class="pre">=</span></span><span class="default_value"><span class="pre">list()</span></span></em><span class="sig-paren">)</span><a class="reference internal" href="../../../_modules/micom/workflows/db_media.html#complete_db_medium"><span class="viewcode-link"><span class="pre">[source]</span></span></a><a class="headerlink" href="#micom.workflows.complete_db_medium" title="Permalink to this definition">#</a></dt>
<dd><p>Complete a growth medium for all models in a database.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters<span class="colon">:</span></dt>
<dd class="field-odd"><ul class="simple">
<li><p><strong>model_db</strong> (<em>str</em>) – A pre-built model database. If ending in <cite>.qza</cite> must be a Qiime 2
artifact of type <cite>MetabolicModels[JSON]</cite>. Can also be a folder,
zip (must end in <cite>.zip</cite>) file or None if the taxonomy contains a
column <cite>file</cite>.</p></li>
<li><p><strong>medium</strong> (<em>pd.DataFrame</em>) – A growth medium. Must have columns “reaction” and “flux” denoting
exchange reactions and their respective maximum flux. Can not be sample
specific.</p></li>
<li><p><strong>growth</strong> (<em>positive float</em><em> or </em><em>pandas.Series</em>) – The minimum growth rate the model has to achieve with the (fixed) medium. If
a Series will have a minimum growth rate for each id/taxon in the model db.</p></li>
<li><p><strong>max_added_import</strong> (<em>positive float</em>) – Maximum import flux for each added additional import not included in the growth
medium. If positive will expand the medium with additional imports in order to
fulfill the growth objective.</p></li>
<li><p><strong>minimize_components</strong> (<em>boolean</em>) – Whether to minimize the number of components instead of the total
import flux. Might be more intuitive if set to True but may also be
slow to calculate.</p></li>
<li><p><strong>weights</strong> (<em>str</em>) – Will scale the fluxes by a weight factor. Can either be “mass” which will
scale by molecular mass, a single element which will scale by
the elemental content (for instance “C” to scale by carbon content).
If None every metabolite will receive the same weight.
Will be ignored if <cite>minimize_components</cite> is True.</p></li>
<li><p><strong>threads</strong> (<em>int &gt;=1</em>) – The number of parallel workers to use when building models. As a
rule of thumb you will need around 1GB of RAM for each thread.</p></li>
<li><p><strong>strict</strong> (<em>list</em>) – Whether to match the imports in the predefined medium exactly. For reactions IDs
listed here will not allow additional import of the components in the provided
medium. For example, if your input medium has a flux of 10 mmol/(gDW*h) defined
and the requested growth rate can only be fulfilled by ramping this up that
would be allowed in non-strict mode but forbidden in strict mode. To match all
medium components to strict mode use <cite>strict=medium.global_id</cite>.</p></li>
</ul>
</dd>
<dt class="field-even">Returns<span class="colon">:</span></dt>
<dd class="field-even"><p>Returns an annotated manifest file with a column <cite>can_grow</cite> that tells you
whether the model can grow on the (fixed) medium, and a column <cite>added</cite> that
gives the number of added imports apart from the ones in the medium.</p>
</dd>
<dt class="field-odd">Return type<span class="colon">:</span></dt>
<dd class="field-odd"><p>tuple of (manifest, import fluxes)</p>
</dd>
</dl>
</dd></dl>

</section>
</section>
</section>

        </article>
      </div>
      <footer>
        
        <div class="related-pages">
          
          
        </div>
        <div class="bottom-of-page">
          <div class="left-details">
            <div class="copyright">
                Copyright &#169; 2017-2021, Christian Diener
            </div>
            Made with <a href="https://www.sphinx-doc.org/">Sphinx</a> and <a class="muted-link" href="https://pradyunsg.me">@pradyunsg</a>'s
            
            <a href="https://github.com/pradyunsg/furo">Furo</a>
            
          </div>
          <div class="right-details">
            
          </div>
        </div>
        
      </footer>
    </div>
    <aside class="toc-drawer">
      
      
      <div class="toc-sticky toc-scroll">
        <div class="toc-title-container">
          <span class="toc-title">
            On this page
          </span>
        </div>
        <div class="toc-tree-container">
          <div class="toc-tree">
            <ul>
<li><a class="reference internal" href="#"><code class="xref py py-mod docutils literal notranslate"><span class="pre">micom.workflows</span></code></a><ul>
<li><a class="reference internal" href="#submodules">Submodules</a></li>
<li><a class="reference internal" href="#package-contents">Package Contents</a><ul>
<li><a class="reference internal" href="#functions">Functions</a></li>
<li><a class="reference internal" href="#attributes">Attributes</a><ul>
<li><a class="reference internal" href="#micom.workflows.workflow"><code class="docutils literal notranslate"><span class="pre">workflow()</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.GrowthResults"><code class="docutils literal notranslate"><span class="pre">GrowthResults</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.save_results"><code class="docutils literal notranslate"><span class="pre">save_results()</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.load_results"><code class="docutils literal notranslate"><span class="pre">load_results()</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.build"><code class="docutils literal notranslate"><span class="pre">build()</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.build_database"><code class="docutils literal notranslate"><span class="pre">build_database()</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.grow"><code class="docutils literal notranslate"><span class="pre">grow()</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.tradeoff"><code class="docutils literal notranslate"><span class="pre">tradeoff()</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.fix_medium"><code class="docutils literal notranslate"><span class="pre">fix_medium()</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.minimal_media"><code class="docutils literal notranslate"><span class="pre">minimal_media()</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.check_db_medium"><code class="docutils literal notranslate"><span class="pre">check_db_medium()</span></code></a></li>
<li><a class="reference internal" href="#micom.workflows.complete_db_medium"><code class="docutils literal notranslate"><span class="pre">complete_db_medium()</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>

          </div>
        </div>
      </div>
      
      
    </aside>
  </div>
</div><script data-url_root="../../../" id="documentation_options" src="../../../_static/documentation_options.js"></script>
    <script src="../../../_static/doctools.js"></script>
    <script src="../../../_static/sphinx_highlight.js"></script>
    <script src="../../../_static/scripts/furo.js"></script>
    <script crossorigin="anonymous" integrity="sha256-Ae2Vz/4ePdIu6ZyI/5ZGsYnb+m0JlOmKPjt6XZ9JJkA=" src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    </body>
</html>